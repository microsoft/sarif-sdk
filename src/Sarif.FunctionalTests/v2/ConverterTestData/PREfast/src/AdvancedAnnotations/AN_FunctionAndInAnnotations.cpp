
#include <sal.h>

void UseValues(_In_ int * input);

void CallUseValues(bool flag)
{
    int source;
    if (flag)
        UseValues(&source);
}
