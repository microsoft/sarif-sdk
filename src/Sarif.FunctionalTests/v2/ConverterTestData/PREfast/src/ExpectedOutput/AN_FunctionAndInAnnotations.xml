<?xml version='1.0' encoding='UTF-8'?>
<DEFECTS><DEFECT><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedannotations\</FILEPATH><FILENAME>an_functionandinannotations.cpp</FILENAME><LINE>10</LINE><COLUMN>17</COLUMN></SFA><DEFECTCODE>6001</DEFECTCODE><DESCRIPTION>Using uninitialized memory 'source'.</DESCRIPTION><FUNCTION>CallUseValues</FUNCTION><DECORATED>?CallUseValues@@YAX_N@Z</DECORATED><FUNCLINE>6</FUNCLINE><PROBABILITY>1</PROBABILITY><RANK>3</RANK><CATEGORY><RULECATEGORY>mspft</RULECATEGORY></CATEGORY><PATH><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedannotations\</FILEPATH><FILENAME>an_functionandinannotations.cpp</FILENAME><LINE>8</LINE><COLUMN>8</COLUMN><KEYEVENT><ID>1</ID><KIND>declaration</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'source' is not initialized</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedannotations\</FILEPATH><FILENAME>an_functionandinannotations.cpp</FILENAME><LINE>9</LINE><COLUMN>8</COLUMN><KEYEVENT><ID>2</ID><KIND>branch</KIND><IMPORTANCE>Full</IMPORTANCE><MESSAGE>Enter this branch, (assume 'flag')</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedannotations\</FILEPATH><FILENAME>an_functionandinannotations.cpp</FILENAME><LINE>10</LINE><COLUMN>17</COLUMN><KEYEVENT><ID>3</ID><KIND>usage</KIND><IMPORTANCE>Full</IMPORTANCE><MESSAGE>'source' is an Input to 'UseValues' (declared on line 4)    (Stmt: UseValues(&amp;source))</MESSAGE></KEYEVENT></SFA><SFA><FILEPATH>e:\enlistcsa\sources\wincsa\src\ddsuites\src\vs\edtqa\static\xmlvalidation\advancedannotations\</FILEPATH><FILENAME>an_functionandinannotations.cpp</FILENAME><LINE>10</LINE><COLUMN>17</COLUMN><KEYEVENT><ID>4</ID><KIND>usage</KIND><IMPORTANCE>Essential</IMPORTANCE><MESSAGE>'source' is used, but may not have been initialized</MESSAGE></KEYEVENT></SFA></PATH></DEFECT></DEFECTS>